src/lib/siwe.ts

```diff
@@ -4,8 +4,9 @@ import { getSigner } from "./util.js";
 
 export async function siwe(this: Connection): Promise<Token> {
   this.signer = this.signer ?? (await getSigner());
-  const chainId = this.options.chainId;
 
-  this.token = await userCreatesToken(this.signer, chainId);
+  await this.checkNetwork();
```

Comment:
Waiting to check that the chains match until we create the token.

Reply:
Yeh killer, more laziness!